mmpose/models/backbones/vit.py

```diff
@@ -0,0 +1,429 @@
+                    feedforward_channels=mlp_ratio * embed_dims,
+                    attn_drop_rate=attn_drop_rate,
+                    drop_rate=drop_rate,
+                    drop_path_rate=dpr[i],
+                    num_fcs=num_fcs,
+                    qkv_bias=qkv_bias,
+                    act_cfg=act_cfg,
+                    norm_cfg=norm_cfg,
+                    with_cp=with_cp,
+                    batch_first=True))
+
+        self.final_norm = final_norm
+        if final_norm:
+            _, self.last_norm = build_norm_layer(norm_cfg, embed_dims)
+        else:
+            self.last_norm = nn.Identity()
+
+    def init_weights(self, pretrained=None):
+        """Initialize the weights in backbone.
+
+        Args:
+            pretrained (str, optional): Path to pre-trained weights.
+                Defaults to None.
+        """
+        if isinstance(pretrained, str):
+            logger = get_root_logger()
+            ckpt = _load_checkpoint(
+                pretrained, logger=logger, map_location='cpu')
+            if 'state_dict' in ckpt:
+                _state_dict = ckpt['state_dict']
+            elif 'model' in ckpt:
+                _state_dict = ckpt['model']
+            else:
+                _state_dict = ckpt
+
+            state_dict = OrderedDict()
+            for k, v in _state_dict.items():
+                if k.startswith('backbone.'):
+                    state_dict[k[9:]] = v
+                else:
+                    state_dict[k] = v
+
+            # strip prefix of state_dict
+            if list(state_dict.keys())[0].startswith('module.'):
+                state_dict = {k[7:]: v for k, v in state_dict.items()}
+
+            # convert the weights from MAE to mmcls
+            new_ckpt = OrderedDict()
+
+            for k, v in state_dict.items():
+                if k.startswith('blocks'):
+                    new_key = k.replace('blocks', 'layers')
+                    if 'norm' in new_key:
+                        new_key = new_key.replace('norm', 'ln')
+                    elif 'mlp.fc1' in new_key:
+                        new_key = new_key.replace('mlp.fc1', 'ffn.layers.0.0')
+                    elif 'mlp.fc2' in new_key:
+                        new_key = new_key.replace('mlp.fc2', 'ffn.layers.1')
+                    elif 'attn.qkv' in new_key:
+                        new_key = new_key.replace('attn.qkv.',
+                                                  'attn.attn.in_proj_')
+                    elif 'attn.proj' in new_key:
+                        new_key = new_key.replace('attn.proj',
+                                                  'attn.attn.out_proj')
+                    new_ckpt[new_key] = v
+                elif k.startswith('patch_embed'):
+                    new_key = k.replace('patch_embed.proj',
+                                        'patch_embed.projection')
+                    new_ckpt[new_key] = v
+                else:
```

Comment:
The downloaded backbone has keys `norm.weight` and `norm.bias`, but in the model the two are called `last_norm.weight` and `last_norm.bias`, it is necessary to add another conversion?

Reply:
No need. The last norm layer is re-initialized for the pose dataset.